blocks/bricks/base.py

```diff
@@ -331,14 +337,18 @@ def __getattr__(self, name):
         # These always belong to the parent (the unbound application)
         if name in ('delegate_function', 'properties'):
             return getattr(self.application, name)
+        if name in self.properties.values():
+            return getattr(self.application.brick, name)
         if name in self.properties:
-            return self.properties[name](self.brick)
+            return getattr(self, self.properties[name])(self.brick)
         # First try the parent (i.e. class level), before trying the delegate
         try:
             return getattr(self.application, name)
         except AttributeError:
             if self.delegate_function:
-                return getattr(self.delegate_function(self.brick), name)
+                return getattr(getattr(self.brick,
+                                       self.delegate_function)(),
```

Comment:
I am confused: do you assume that the delegate application is always an application of the same brick? But this is not true, see a quote from `sequence_generators.py`:

```python
    @compute_states.delegate
    def compute_states_delegate(self):
        return self.transition.apply
```

I honestly do not know how it works now, and moreover think that we should add a test for that.